### PR TITLE
Moved Table to Separate components and Moved controls to PanelControls Component

### PR DESCRIPTION
**Pull request by vrackovic**
- Moved Table component render from PanelContainerTwoCells to Formula and Category as separate calls
- Outsourced panel controls to separate component
- Outsourced tableData to parent component

### CODE REVIEW

File: web/src/main/frontend/config-overrides.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const { injectBabelPlugin } = require("react-app-rewired");
+const rewireMobX = require("react-app-rewire-mobx");
+
+module.exports = function override(config, env) {
+  config = injectBabelPlugin("babel-plugin-styled-components", config);
+  config = rewireMobX(config, env);
+
+  return config;
+};
```
Comment (iscekic) [inline web/src/main/frontend/config-overrides.js]: Missing newline - was this file formatted with prettier?

File: web/src/main/frontend/src/base/Store.js [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+    {title: 'Web Test', text: 'Test containing JS, HTML and CSS questions'},
+    {title: 'C# Test', text: 'Test containing C# and OOP questions'},
+    {title: 'Java Test', text: 'Test containing Java and OOP questions'},
+    {title: 'Android Test', text: 'Test containing...'}
+  ];
+
+  @computed
+  get category() {
+    return this.categories.map(c => {
+      c.id = Math.floor(Math.random() * 1000);
```
Comment (iscekic) [inline web/src/main/frontend/src/base/Store.js]: Don't mutate the original array in `map`. Return a completely new item. :smile:

File: web/src/main/frontend/src/base/Store.js [hunk 3]
```diff
@@ -1,3 +1,40 @@
...
+      return c;
+    });
+  }
+
+  @computed
+  get formula() {
+    return this.formulas.map(f => {
+      f.id = Math.floor(Math.random() * 1000);
```
Comment (iscekic) [inline web/src/main/frontend/src/base/Store.js]: Same here.

File: web/src/main/frontend/src/components/ui/AddButton.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+const button = props => {
+  return (
+    <div>
+      <div href={props.href} className={props.styling}>
```
Comment (iscekic) [inline web/src/main/frontend/src/components/ui/AddButton.js]: Rename `styling` to `className`. If this isn't possible because `className` is special, rename it to something else, eg. `cssClass`.

File: web/src/main/frontend/src/components/ui/Button.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+
+const button = props => {
```
Comment (iscekic) [inline web/src/main/frontend/src/components/ui/Button.js]: Why do we have `Button` and `AddButton` as separate components? They are mostly the same.

File: web/src/main/frontend/src/components/ui/Table.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+class Table extends React.Component {
+  render() {
+    const tableRows = [];
```
Comment (iscekic) [inline web/src/main/frontend/src/components/ui/Table.js]: `tableRows` can be a computed class property.

File: web/src/main/frontend/package.json [hunk 7]
```diff
@@ -3,6 +3,7 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "babel-plugin-styled-components": "^1.5.1",
```
Comment (iscekic) [inline web/src/main/frontend/package.json]: Why was this added? I don't see any `styled-components`. :smile:

File: web/src/main/frontend/config-overrides.js (hunk 1 above)
Comment (vrackovic) [reply web/src/main/frontend/config-overrides.js]: Don't think so. Newline will be added.

File: web/src/main/frontend/package.json (hunk 7 above)
Comment (vrackovic) [reply web/src/main/frontend/package.json]: It will be removed :D

File: web/src/main/frontend/src/components/ui/Button.js (hunk 5 above)
Comment (vrackovic) [reply web/src/main/frontend/src/components/ui/Button.js]: For now i solved it with combining both components, and render different things via ternary operator.

Comment (iscekic) [reply web/src/main/frontend/src/components/ui/Button.js]: There should be no ternary operator. Have the component render a button based on props, and handle everything else in the parent.

File: web/src/main/frontend/src/base/Store.js [hunk 8]
```diff
@@ -1,3 +1,46 @@
...
+    {title: 'Web Test', text: 'Test containing JS, HTML and CSS questions'},
+    {title: 'C# Test', text: 'Test containing C# and OOP questions'},
+    {title: 'Java Test', text: 'Test containing Java and OOP questions'},
+    {title: 'Android Test', text: 'Test containing...'}
+  ];
+
+  @computed
+  get category() {
+    return this.categories.map(c => {
+      let computedCategory = {
```
Comment (iscekic) [inline web/src/main/frontend/src/base/Store.js]: I'm assuming we're generating a random id to use as `key`? Do these objects not have a unique property which can be used?

If not, extract this callback into a separate function, eg.

```js
const addRandomId = obj => ({ ...obj, id: Math.floor(Math.random() * 1000) });

// elsewhere
@computed
get category() {
  return this.categories.map(addRandomId);
}
```

File: web/src/main/frontend/src/components/ui/Table.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+
+    return (
+      <table>
+        <thead>
+          <tr>
+            <th scope="col">{this.props.heading}</th>
+            <th scope="col">Description</th>
+          </tr>
+        </thead>
+        {this.props.tableData ? tableData : null}
```
Comment (iscekic) [inline web/src/main/frontend/src/components/ui/Table.js]: No need for this, just `{tableData}` should be enough.

File: web/src/main/frontend/src/base/Store.js (hunk 8 above)
Comment (vrackovic) [reply web/src/main/frontend/src/base/Store.js]: There are two objects with same `title` property, and it was used for assigning `key`-s.
